cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/ReplicaDistributionGoal.java

```diff
@@ -81,24 +82,36 @@ public ReplicaDistributionGoal(BalancingConstraint balancingConstraint) {
    * To avoid churns, we add a balance margin to the user specified rebalance threshold. e.g. when user sets the
    * threshold to be {@link BalancingConstraint#replicaBalancePercentage()}, we use
    * ({@link BalancingConstraint#replicaBalancePercentage()}-1)*{@link #BALANCE_MARGIN} instead.
+   *
+   * @param optimizationOptions Options to adjust balance percentage with margin in case goal optimization is triggered
+   * by goal violation detector.
    * @return the rebalance threshold with a margin.
    */
-  private double balancePercentageWithMargin() {
-    return (_balancingConstraint.replicaBalancePercentage() - 1) * BALANCE_MARGIN;
+  private double balancePercentageWithMargin(OptimizationOptions optimizationOptions) {
+    double balancePercentage = optimizationOptions.isTriggeredByGoalViolation()
+                               ? _balancingConstraint.replicaBalancePercentage()
+                                 * _balancingConstraint.goalViolationDistributionThresholdMultiplier()
+                               : _balancingConstraint.replicaBalancePercentage();
+
```

Comment:
maybe we can change to 
`_balancingConstraint.replicaBalancePercentage() * (optimizationOptions.isTriggeredByGoalViolation() ? _balancingConstraint.goalViolationDistributionThresholdMultiplier() : 1)` 
to reduce a call to `_balancingConstraint.replicaBalancePercentage()`

Reply:
I am confused -- this does not reduce the call to `_balancingConstraint.replicaBalancePercentage()`, we already call it just once.
Would you please clarify?

Reply:
you are right, it is an `if-else` branch, maybe it make code more concise.